Doesn't build with GHC-8.8.1/base-4.13

```
Build profile: -w ghc-8.8.1 -O0
In order, the following will be built (use -v for more details):
 - sqlite-simple-0.4.16.0 (lib) (first run)
Configuring library for sqlite-simple-0.4.16.0..
Preprocessing library for sqlite-simple-0.4.16.0..
Building library for sqlite-simple-0.4.16.0..
[ 1 of 11] Compiling Database.SQLite.Simple.Ok ( Database/SQLite/Simple/Ok.hs, /code/mess/sqlite-simple/sqlite-simple-0.4.16.0/.dist-newstyle-trustee/5e7d9fd33a5064be14b944e0c13b8f4b86cbfea5c40b9a2fedcf90f6574fd737601d0660b79d714d2285991de221c3fd383f34af92592b8382c7510520aa0702/build/x86_64-linux/ghc-8.8.1/sqlite-simple-0.4.16.0/noopt/build/Database/SQLite/Simple/Ok.o )


Database/SQLite/Simple/Ok.hs:75:5: error:
    ‘fail’ is not a (visible) method of class ‘Monad’
   |
75 |     fail str = Errors [SomeException (ErrorCall str)]
   |     ^^^^
```

```
Build profile: -w ghc-8.8.1 -O0
In order, the following will be built (use -v for more details):
 - sqlite-simple-0.4.15.0 (lib) (first run)
Configuring library for sqlite-simple-0.4.15.0..
Preprocessing library for sqlite-simple-0.4.15.0..
Building library for sqlite-simple-0.4.15.0..
[ 1 of 10] Compiling Database.SQLite.Simple.Ok ( Database/SQLite/Simple/Ok.hs, /code/mess/sqlite-simple/sqlite-simple-0.4.15.0/.dist-newstyle-trustee/5e7d9fd33a5064be14b944e0c13b8f4b86cbfea5c40b9a2fedcf90f6574fd737601d0660b79d714d2285991de221c3fd383f34af92592b8382c7510520aa0702/build/x86_64-linux/ghc-8.8.1/sqlite-simple-0.4.15.0/noopt/build/Database/SQLite/Simple/Ok.o )


Database/SQLite/Simple/Ok.hs:75:5: error:
    ‘fail’ is not a (visible) method of class ‘Monad’
   |
75 |     fail str = Errors [SomeException (ErrorCall str)]
   |     ^^^^
```

I also made revisions to affected versions on Hackage

- https://hackage.haskell.org/package/sqlite-simple-0.4.16.0/revisions/ 
- https://hackage.haskell.org/package/sqlite-simple-0.4.15.0/revisions/

--- a/Database/SQLite/Simple/Ok.hs
+++ b/Database/SQLite/Simple/Ok.hs
@@ -1,3 +1,4 @@
+{-# LANGUAGE CPP                #-}
 {-# LANGUAGE DeriveDataTypeable #-}
 {-# LANGUAGE DeriveFunctor      #-}
 
@@ -31,8 +32,12 @@ module Database.SQLite.Simple.Ok where
 
 import Control.Applicative
 import Control.Exception
-import Control.Monad(MonadPlus(..))
+import Control.Monad (MonadPlus(..))
 import Data.Typeable
+
+#if !MIN_VERSION_base(4,13,0) && MIN_VERSION_base(4,9,0)
+import Control.Monad.Fail
+#endif
 
 -- FIXME:   [SomeException] should probably be something else,  maybe
 --          a difference list (or a tree?)
@@ -72,8 +77,10 @@ instance Monad Ok where
     Errors es >>= _ = Errors es
     Ok a      >>= f = f a
 
+#if MIN_VERSION_base(4,9,0)
 instance MonadFail Ok where
     fail str = Errors [SomeException (ErrorCall str)]
+#endif
 
 -- | a way to reify a list of exceptions into a single exception
 

--- a/Database/SQLite/Simple/Types.hs
+++ b/Database/SQLite/Simple/Types.hs
@@ -1,4 +1,4 @@
-{-# LANGUAGE DeriveDataTypeable, DeriveFunctor, GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE DeriveDataTypeable, DeriveFunctor, GeneralizedNewtypeDeriving, CPP #-}
 
 ------------------------------------------------------------------------------
 -- |
@@ -28,6 +28,10 @@ import           Data.String (IsString(..))
 import           Data.Tuple.Only (Only(..))
 import           Data.Typeable (Typeable)
 import qualified Data.Text as T
+
+#if !MIN_VERSION_base(4,11,0)
+import Data.Semigroup
+#endif
 
 -- | A placeholder for the SQL @NULL@ value.
 data Null = Null

--- a/sqlite-simple.cabal
+++ b/sqlite-simple.cabal
@@ -51,7 +51,7 @@ Library
     bytestring >= 0.9,
     containers,
     direct-sqlite >= 2.3.13 && < 2.4,
-    semigroups == 0.18.*,
+    semigroups >= 0.18 && < 0.20,
     template-haskell,
     text >= 0.11,
     time,

--- a/stack.yaml
+++ b/stack.yaml
@@ -1,4 +1,9 @@
-resolver: lts-15.1
+resolver: lts-15.1     # 8.8.2  - base-4.13
+# resolver: lts-14.27  # 8.6.5  - base-4.12
+# resolver: lts-12.26  # 8.4.4  - base-4.11
+# resolver: lts-11.22  # 8.2.2  - base-4.10
+# resolver: lts-9.21   # 8.0.2  - base-4.9
+# resolver: lts-6.35   # 7.10.3 - base-4.8
 
 extra-deps:
   - direct-sqlite-2.3.26

--- a/stack.yaml
+++ b/stack.yaml
@@ -1,1 +1,4 @@
-resolver: lts-11.9
+resolver: lts-15.1
+
+extra-deps:
+  - direct-sqlite-2.3.26

--- a/test/ParamConv.hs
+++ b/test/ParamConv.hs
@@ -1,5 +1,8 @@
+{-# LANGUAGE ScopedTypeVariables, OverloadedStrings, CPP #-}
+
+#if MIN_VERSION_base(4,9,0)
 {-# OPTIONS_GHC -Wno-overflowed-literals #-}
-{-# LANGUAGE ScopedTypeVariables, OverloadedStrings #-}
+#endif
 
 module ParamConv (
     testParamConvNull
@@ -240,5 +243,3 @@ testParamNamed TestEnv{..} = TestCase $ do
   executeNamed conn "INSERT INTO np (b) VALUES (:b)" [":b" := True]
   [Only t1] <- query_ conn "SELECT b FROM np"
   True @=? t1
-
-

--- a/test/Simple.hs
+++ b/test/Simple.hs
@@ -1,3 +1,4 @@
+{-# LANGUAGE CPP #-}
 {-# LANGUAGE OverloadedStrings #-}
 
 module Simple (
@@ -25,6 +26,10 @@ import qualified Data.Text.Lazy as LT
 import           Data.Time (UTCTime, Day)
 
 import           Common
+
+#if !MIN_VERSION_base(4,11,0)
+import Data.Monoid ((<>))
+#endif
 
 -- Simplest SELECT
 testSimpleOnePlusOne :: TestEnv -> Test

